### ISSUE TITLE
How to identity significant change points between cell types?

### ISSUE THREAD

**Issue by qoiopipq** [shenkers/isoscm#19]
Part of my project is to find differential change points/poly A sites between thymus and liver cells. These are the cutoffs I used to select significant change points to mark alternative poly A sites:
- change points in terminal exons only
- p-values <= 0.01
- upstream coverage >= lower median value in cell types
- |differential usage| >= 0.2

I've got 230 significant change points that associated with 223 genes (out of a few thousands of change points/genes in the unfiltered data, which is great!). Then I did GO analysis to narrow down the list. But I came across some genes with identified change points that having very similar 3' exons (in lengths), I've included Ptbp1 gene as an example (red dash line is the identified change point, yellow dash line is the end of downstream segment, scale of reads in thymus (1st bam file) is 0-1500, in liver (2nd bam file) is 0-500):
![ptbp1](https://cloud.githubusercontent.com/assets/16911467/14843961/6a3a508e-0c97-11e6-954f-7d84b76ecb4d.png)

(RefSeq: in black; Thymus exons: in blue; Liver exons: in green)
In this example, the shortest 3' exons in both cell types are identical. The estimated usages of the change point in thymus is 0.845, in liver is 0.358.  The decrease of reads at the identified change point is quite significant in each cell type. Apart from the shortest 3' exons, the longest 3' exons and the 3' exons with intermediate lengths in both cell types are very similar in lengths. In the situation like this, the shortest 3' exons are found in both cell types, so how to justify that the identified change point is differentially used between cell types instead of among three terminal exons within a single cell type? 

Another question is that the shorter and longer 3’ exons have been identified from compare and diff steps are found within a single cell type instead of between cell types. 
![trim8](https://cloud.githubusercontent.com/assets/16911467/14843973/8643e98e-0c97-11e6-9351-0d6ccdbf369f.png)
Trim 8  is one of these genes (red dash line is the identified change point, yellow dash line is the end of downstream segment), the first 3’ exon in thymus is the extended 3’ utr, and the third 3’ exon in thymus is the shorter 3’ utr. Both of them are found in thymus, any ideas about how to exclude these significant change points in my results?

Thanks!

**Comment by shenkers** [comment]
Hi @qoiopipq,

There are several questions here, I'll try to address each below.

**"I came across some genes with identified change points that having very similar 3' exons (in lengths), I've included Ptbp1 gene as an example"**

The `assemble` step will attempt to construct gene models for a particular sample, ie looking specifically at the liver or thymus sample in your case. Since the RNA-seq coverage can be different from sample to sample, the assembled 3'UTRs can also be different, depending on what the input RNA-seq looked like. In the case of Ptbp1 you show, IsoSCM happened to decide that the short 3'UTR isoform is at the same location between samples, but this will not necessarily be the case in general. If you want to make a comparison between tissues, I would recommend you use the output of the `compare` step. 

The `compare` step takes the output of the two assemblies from the `assemble` step, and uses that to define regions that will be compared between the two samples. The `compare` step will re-run the segmentation looking at both samples simultaneously. The changepoints identified by the `compare` step will not necessarily be the same as those identified by the `assemble` step; the changepoints identified by `compare` is a consensus set of changepoints observed in either sample.

**"how to justify that the identified change point is differentially used between cell types instead of among three terminal exons within a single cell type?"**

The `compare` step should be used to identify changepoints that are differentially used between conditions. I'm not sure what the second alternative you are considering is, could you rephrase this?

**"Another question is that the shorter and longer 3’ exons have been identified from compare and diff steps are found within a single cell type instead of between cell types... any ideas about how to exclude these significant change points in my results?"**

The compare step outputs a quantification of the differential usage at each changepoint. If a changepoint is not differentially used between conditions, the estimated differential usage for that changepoint should be close to 0. Could you include the rows from the `compare` step that correspond to this locus? Is this changepoint estimated to have a |differential usage| > 0.2 between liver and thymus? 

The y-axis scaling for trim8 makes it a little difficult to see the liver sample, so it's difficult to judge the differential expression. Could you please include another snapshot of that locus?

Let me know if there's anything else I can help with.

**Comment by qoiopipq** [comment]
This table contains relevant columns of Ptbp1 and Trim8 genes I got from compare and diff steps, the change points are marked as red dash lines in the previous post.  
![isoscm_question](https://cloud.githubusercontent.com/assets/16911467/14871303/513e6b3c-0d24-11e6-89aa-a8799122339a.png)
(the usage of Ptbp1 in liver is 0.563 not 0.358, which I got it wrong in the previous post)
For Ptbp1, the short 3 isoforms are found at the same location between samples, can I consider it as a rare case? Definitely there are short isoforms in both cell types, but in terms of comparing 3' UTR lengths between cell types, Ptbp1 may not be the gene we're looking for. 

For Trim 8, the scale of reads is 0-500 in both cases, site usage in thymus is 0.970 and 0.769 in liver, which just pass |differential usage| >0.2 cutoff. Reads in liver are much lower than reads in thymus. Is it possible that insufficient reads in liver that results in both short and long isoforms are identified in thymus? Do you think I should add another cutoff of up/downstream coverage to exclude genes with insufficient reads?

Cheers!

**Comment by shenkers** [comment]
sorry for the delayed response. It's hard to tell because of the scaling from the first image you sent what this change-point looks like in the raw data. From the numbers in the table it looks like there is some differential usage. If the events identified as having differential usage do not look reliable at your current strategy, I think using a higher cutoff for upstream coverage is reasonable.
